PIL/ImageDraw.py

```diff
@@ -305,6 +304,120 @@ def multiline_textsize(self, text, font=None, spacing=4):
+    #    The best lineHeight would be that returned from the font.font.height
+    #    attribute, but in versions of PIL in which that isn't accessible,
+    #    sum( font.getmetrics()) is used instead.
+    def textInfo( self, text, font=None, lineHeight=None, lineHeightPercent=None ):
+        if font is None:
+            font = self.getfont()
+        lines = text.split('\n')
+        txMax = 0
+        txMin = 0
+        txWid = 0
+        tyMax = None
+        tyMin = None
+        if len( lines ) > 1:
+            if lineHeight is None:
+                if lineHeightPercent is None:
+                    lineHeightPercent = 100
+                try:
+                    lineHeight = font.font.height
+                except Exception as exc:
+                    lineHeight = sum( font.getmetrics())
+                lineHeight = int( lineHeight * lineHeightPercent / 100 )
+
+        lineBBs = []
+        for line in lines:
+            lxMin, lyMin, lxMax, lyMax = font.getBB( line )
+            lineBBs.append(( lxMin, lyMin, lxMax, lyMax, line ))
+            lxWid = lxMax - lxMin
+            if txWid < lxWid:
+                txWid = lxWid
+            if txMax < lxMax:
+                txMax = lxMax
+            if txMin > lxMin:
+                txMin = lxMin
+            if tyMax is None:
+                tyMax = lyMax # from first line only
+            if tyMin is None:
+                tyMin = 0 # skips first line (unless it is also last)
+            else:
+                tyMin -= lineHeight
+        tyMin += lyMin # from last line
+        return ( txMin, tyMin, txMax, tyMax, txWid, lineHeight, lineBBs )
+
+    # Text is drawn as close as possible to the specified alignment edges of the
+    # image, without truncation on those edges, then adjusted by the origin value.
+    # alignX or alignY of 'exact' means to use the specified origin point exactly
+    # in that direction. Otherwise origin is used as an offset from the calculated
+    # alignment position. alignX can also be 'left', 'center', or 'right'; alignY
+    # can also be 'top', 'middle', or 'bottom'. justifyX can be 'left', 'center',
+    # or 'right'. Other parameters like textInfo.
+    def textAtPos( self, text, font=None, lineHeight=None, lineHeightPercent=None,
+                   origin=( 0, 0 ), alignX='exact', alignY='exact', justifyX='left',
+                   fill=None ):
+        if font is None:
+            font = self.getfont()
+        ink, fill = self._getink( fill )
+        if ink is None:
+            inx = fill
+            if ink is None:
+                return
+        if justifyX not in ('left', 'center', 'right'):
+            raise ValueError('Unknown justifyX value "%s".' % justifyX )
+        txMin, tyMin, txMax, tyMax, txWid, lineHeight, lineBBs = self.textInfo(
+            text, font, lineHeight, lineHeightPercent )
+        if alignX == 'exact':
+            ox = 0
```

Comment:
What is the `o` in `ox` and `oy`? Can they be named more descriptively (in snake_case)?